includes/class-server-rules.php

```
@@ -0,0 +1,117 @@
+			if ( false !== strpos( $server_software, 'nginx' ) ) {
+				return 'nginx';
+			}
+
+			// Fallback: check SAPI or common environment variables.
+			if ( 'fpm-fcgi' === php_sapi_name() ) {
+				// Nginx usually uses FPM.
+				return 'nginx';
+			}
+
+			return 'other';
+		}
```

PHPMD (2.15.0)

[error] get_server_type accesses the super-global variable $_SERVER. (undefined)

(Superglobals)